openprescribing/api/views_spending.py

```diff
@@ -442,20 +441,17 @@ def _get_total_prescribing_entries(bnf_code_prefixes):
     # powered by the `vw__presentation_summary` table which summed over all
     # practice types)
     group_all = get_row_grouper("all_practices")
-    items_matrix = group_all.sum(items_matrix)
-    quantity_matrix = group_all.sum(quantity_matrix)
-    actual_cost_matrix = group_all.sum(actual_cost_matrix)
+    items_matrix = group_all.sum_one_group(items_matrix)
+    quantity_matrix = group_all.sum_one_group(quantity_matrix)
+    actual_cost_matrix = group_all.sum_one_group(actual_cost_matrix)
     # Yield entries for each date (unlike _get_prescribing_entries below we
     # return a value for each date even if it's zero as this is what the
     # original API did)
-    for date, col_offset in sorted(db.date_offsets.items()):
-        # The grouped matrices only ever have one row (which represents the
-        # total over all practices) so we always want row 0 in our index
-        index = (0, col_offset)
+    for date in items_matrix.labels:
         yield {
-            "items": items_matrix[index],
-            "quantity": quantity_matrix[index],
-            "actual_cost": round(actual_cost_matrix[index], 2),
+            "items": items_matrix[date],
```

Comment:
Actually, my intention was that `__getitem__` would look up values. So for a n-dimensional array it would take a n-tuple and, as a convenience, for a 1-dimensional array it would accept a single index value. In other words, it would mimic numpy's `__getitem__` minus all the slicing stuff.

I've realised that the example above is unnecessarily confusing because I've reused variable names:
```python
items_matrix = group_all.sum_one_group(items_matrix)
```
After the `sum_one_group` call, `items_matrix` is now a 1-dimensional array indexed by date.